### ISSUE TITLE
defualtOptions are overriden by undefined and null

### ISSUE THREAD

**Issue by asafigan** [apollographql/apollo-client#3717]
**Intended outcome:**
`client.watchQuery({fetchPolicy: null})` should use the fetchPolicy in defaultOptions

**Actual outcome:**
Instead it sets it to cache-first because it is the default behavior.

**How to reproduce the issue:**
https://codesandbox.io/s/8zx2m3759l
Watch your network traffic. There should be two request to the graphql endpoint but there is only one.

**Versions**

apolloclient@2.3.5

**Comment by dmarkow** [comment]
Ran into this today. It looks like React Apollo's `<Query />` component works around this by compacting the options before passing them to apollo-client, but not all third party libraries using apollo-client will know about this (I ran into this with [react-apollo-hooks](https://github.com/trojanowski/react-apollo-hooks/issues/76) today).

It seems that apollo-client compacting options before merging into `defaultOptions` would be a better solution rather than expecting other tools to know not to pass in undefined/null options?
